### ISSUE TITLE
'MyCMSPlugin' object has no attribute '_placeholder_cache'

### ISSUE THREAD

**Issue by Chive** [aldryn/aldryn-newsblog#216]
The function [``aldryn_newsblog.models#update_search_index``](https://github.com/aldryn/aldryn-newsblog/blob/develop/aldryn_newsblog/models.py#L515-L529) defines the following code:

```python
@receiver(post_save)
def update_seach_index(sender, instance, **kwargs):
    """
    Upon detecting changes in a plugin used in an Article's content
    (PlaceholderField), update the article's search_index so that we can
    perform simple searches even without Haystack, etc.
    """
    if issubclass(instance.__class__, CMSPlugin):
        placeholder = instance._placeholder_cache
        if hasattr(placeholder, '_attached_model_cache'):
            if placeholder._attached_model_cache == Article:
                article = placeholder._attached_model_cache.objects.get(
                    content=placeholder.pk)
                article.search_data = article.get_search_data(instance.language)
                article.save()
```

In the second line the plugin's placeholder is retrieved by doing ``placeholder = instance._placeholder_cache`` which fails miserably (``'MyCMSPlugin' object has no attribute '_placeholder_cache'``) when I am trying to work with some custom plugins of mine.

I am now wondering - is ``_placeholder_cache`` a "standard" attribute of/for ``CMSPlugin`` objects? The only use within the CMS I can find is in [``cms.models.pagemodel.Page#get_placeholders``](https://github.com/divio/django-cms/blob/develop/cms/models/pagemodel.py#L846-L849) which doesn't seem related.

**Comment by Chive** [comment]
**Update:**

After some fiddling - I found the following:
```bash
ipdb> instance._placeholder_cache
*** AttributeError: 'MyCMSPlugin' object has no attribute '_placeholder_cache'
ipdb> instance.placeholder
<Placeholder: content>
ipdb> instance._placeholder_cache
<Placeholder: content>
```

So it seems like that it's actually being set somewhere. And it's also worth mentioning that it is also set on every following request.

**Comment by Chive** [comment]
Fixed in #217
